neural_nets/activations/activations.py

```diff
@@ -128,3 +128,22 @@ def grad(self, x):
 
     def grad2(self, x):
         return np.zeros_like(x)
+
+
+class Elu(ActivationBase):
+    def __init__(self, alpha=1.0):
+        self.alpha = alpha
+        super().__init__()
+
+    def __str__(self):
+        return "Elu(alpha={})".format(self.alpha)
+
+    def fn(self, z):
+        _z = z * (z > 0) + self.alpha * (np.exp(z) - 1.) * (z < 0)
+        return _z
+
+    def grad(self, x):
+        return 1 if x >= 0 else self.fn(x) + self.alpha
```

Comment:
```suggestion
        return np.where(x >= 0, np.ones_like(x), self.fn(x) + self.alpha)
```
The `else` clause doesn't know what to do with a boolean array. I'd opt for using `np.where` here.